docs/coordinates/inplace.rst

```diff
@@ -0,0 +1,35 @@
+In Place Modification of Coordinates
+====================================
+
+Coordinates are generally considered to be immutable. If you want to create
+another coordinate frame with different data you should use
+`~astropy.coordinates.BaseCoordinateFrame.realize_frame`, this is the safest way
+to change the data in a frame as it creates a new frame with that data.
+
+Due to the fact that this creates a new frame it is quite slow, and in some
+situations it is required that the data is replaced inside a frame with minimal
+changes to the frame for maximum speed. This modification can be done by
+modifiying the values of the representation data as follows::
+
+    c = SkyCoord([1,2],[3,4], unit='deg')
+    c.data.lon[:] = [10, 20] * u.deg
```

Comment:
Use `c.data.lon[()] = ..` since this syntax also works for a scalar (which is a 0-d array internally).